App/Logic/DataUploadLogic.swift

```diff
@@ -232,22 +266,181 @@ extension Logic.DataUpload {
+      let state = context.getState()
+
+      guard !state.ingestion.dummyTrafficSequenceScheduledInSession else {
+        // There is already another sequence schedule for this session.
+        return
+      }
+
+      // Reset the flag in case this is a new session
+      try context.awaitDispatch(SetDummyTrafficSequenceCancelled(value: false))
+
+      guard state.ingestion.dummyTrafficOpportunityWindow.contains(context.dependencies.now()) else {
+        // Not within the opportunity window. Nothing to do.
+        return
+      }
+
+      // Prevent other ingestion sequences from being scheduled in this session
+      try context.awaitDispatch(SetDummyIngestionSequenceScheduledForThisSession(value: true))
+
+      let startDelay = context.dependencies.exponentialDistributionGenerator
+        .exponentialRandom(with: state.configuration.dummyIngestionAverageStartUpDelay)
+
+      // Dispatch a delayed action and return
+      context.dispatch(StartIngestionSequenceIfNotCancelled().deferred(of: startDelay))
+    }
+  }
+
+  /// Attempts a simulated sequence of ingestion requests, unless it was cancelled for outside in the meanwhile.
+  /// -seeAlso: https://github.com/immuni-app/immuni-documentation/blob/master/Traffic%20Analysis%20Mitigation.md
+  struct StartIngestionSequenceIfNotCancelled: AppSideEffect {
+    func sideEffect(_ context: SideEffectContext<AppState, AppDependencies>) throws {
+      var executions = 0
+      while true {
+        // Fetch the state at every run because it might have been updated in the meanwhile
+        let state = context.getState()
+
+        guard !state.ingestion.isDummyTrafficSequenceCancelled else {
+          // The user signaled the intent of starting a genuine upload.
+          break
+        }
+
+        #warning("Define proper dummy request")
+        try? await(context.dependencies.networkManager.uploadData(body: .dummy(), otp: .init()))
+
+        let diceRoll = context.dependencies.uniformDistributionGenerator.randomNumberBetweenZeroAndOne()
+        let threshold = state.configuration.dummyIngestionRequestProbabilities[safe: executions]
+          ?? state.configuration.dummyIngestionRequestProbabilities.last
+          ?? AppLogger.fatalError("No probabilities defined")
+
+        guard diceRoll < threshold else {
+          // The generated probability was above the threshold.
+          break
+        }
+
+        let interRequestDelay = context.dependencies.exponentialDistributionGenerator
+          .exponentialRandom(with: state.configuration.dummyIngestionAverageRequestWaitingTime)
+
+        // Await for a random time
+        try await(Promise<Void>.deferring(of: interRequestDelay))
+        executions += 1
+      }
+
+      // Allow the scheduling of another session.
+      try context.awaitDispatch(SetDummyIngestionSequenceScheduledForThisSession(value: false))
+
+      // Update the opportunity window
+      try context.awaitDispatch(UpdateDummyTrafficOpportunityWindow())
```

Comment:
make so that this is executed no matter what if the simulation is scheduled

Reply:
Done